Neos.Flow/Classes/Mvc/Routing/Router.php

```diff
@@ -267,9 +276,112 @@ protected function createRoutesFromConfiguration()
             }
             $this->routes[] = $route;
         }
+
         $this->routesCreated = true;
     }
 
+    /**
+     * Emits a signal when the router configuration has been loaded
+     *
+     * This signal can be used to add routes during runtime. In the slot make sure to receive the
+     * $routerConfiguration array by reference so you can alter it.
+     *
+     * @param array $routerConfiguration The router configuration
+     * @return void
+     * @Flow\Signal
+     */
+    protected function emitConfigurationLoaded(array &$routerConfiguration): void
+    {
+    }
+
+    public function configureRouteAnnotatedMethods(array &$routerConfiguration): void
+    {
+        $annotatedRoutes = static::resolveRouteAnnotatedMethods($this->objectManager);
+        $routerConfiguration = array_merge(array_values($annotatedRoutes), $routerConfiguration);
+    }
+
+
+    /**
+     * @param ObjectManagerInterface $objectManager
+     * @Flow\CompileStatic
+     * @return array
+     */
+    protected static function resolveRouteAnnotatedMethods(ObjectManagerInterface $objectManager): array
+    {
+        $annotatedRoutes = [];
+        $reflectionService = $objectManager->get(ReflectionService::class);
+        $controllerClassNames = $reflectionService->getAllSubClassNamesForClass(AbstractController::class);
+        foreach ($controllerClassNames as $controllerClassName) {
+            if ($reflectionService->isClassAbstract($controllerClassName)) {
+                continue;
+            }
+
+            $methodNames = $reflectionService->getMethodsAnnotatedWith($controllerClassName, Flow\Route::class);
+            foreach ($methodNames as $methodName) {
+                if (preg_match('/.*Action$/', $methodName) === 0 || $reflectionService->isMethodPublic($controllerClassName, $methodName) === false) {
+                    continue;
```

Comment:
Skipping other methods silently seems fragile to me

Reply:
Yes, let's throw an exception if a non-Action is annotated. It will only happen at compile-time in production anyway

Reply:
@albe @bwaidelich Covered in next commit - looking for help on how to test a wrongly annotated method, without breaking all other tests (blocking compiling)